plugins/autoid/plugin.js

```diff
@@ -1,8 +1,83 @@
 'use strict';
 
 (function () {
+	var editorHasFocus = false;
+	var commandIsActive = false;
+
 	CKEDITOR.plugins.add('autoid', {
 		init: function (editor) {
+			var self = this;
+
+			if (editor && !editor.config.autoid) {
+				editor.config.autoid = {};
+			}
+			self.settings = editor.config.autoid;
+			if (!self.settings) {
+				self.settings = {};
+			}
+
+			editor.addCommand('autoid', {
+				exec: function (editor) {
+					if (!commandIsActive) {
+						start();
+					} else {
+						stop();
+					}
+				},
+				editorFocus: true
+			});
+
+			editor.on("instanceReady", function () {
+				if (self.settings.autostart !== false) {
+					start()
+				}
+			});
+
+			function start() {
+				editor.getCommand('autoid').setState(CKEDITOR.TRISTATE_ON);
+				commandIsActive = true;
+
+				addAllIds();
+			}
+
+			function stop() {
+				editor.getCommand('autoid').setState(CKEDITOR.TRISTATE_OFF);
+				commandIsActive = false;
```

Comment:
what are you thinking that stop() will do?  Pause the generation of further ids for any new elements?

This might be a needless question.  Once we deploy this I'm not sure of when we'd turn it off.